docker.go

```diff
@@ -46,10 +46,73 @@ type Parameters struct {
 
 // Volume is the docker volume details associated to the container
 type Volume struct {
-	ContainerPath string          `json:"containerPath,omitempty"`
-	HostPath      string          `json:"hostPath,omitempty"`
-	External      *ExternalVolume `json:"external,omitempty"`
-	Mode          string          `json:"mode,omitempty"`
+	ContainerPath string            `json:"containerPath,omitempty"`
+	HostPath      string            `json:"hostPath,omitempty"`
+	External      *ExternalVolume   `json:"external,omitempty"`
+	Mode          string            `json:"mode,omitempty"`
+	Persistent    *PersistentVolume `json:"persistent,omitempty"`
+}
+
+type PersistentVolumeType string
+
+const (
+	PersistentVolumeTypeRoot  PersistentVolumeType = "root"
+	PersistentVolumeTypePath  PersistentVolumeType = "path"
+	PersistentVolumeTypeMount PersistentVolumeType = "mount"
+)
+
+// PersistentVolume declares a Volume to be Persistent, and sets
+// the size (in MiB) and optional type, max size (MiB) and constraints for the Volume.
+type PersistentVolume struct {
+	Type        *PersistentVolumeType `json:"type,omitempty"`
+	Size        int                   `json:"size,omitempty"`
```

Comment:
FWIW, the REST documentation [I cited in the original PR](https://github.com/gambol99/go-marathon/pull/200/files#r80254464) does list zero as the minimum value. Could you double-check/test this?

If the size must really be positive, we should still keep the `omitempty`. Otherwise, users will always have to set the existing value explicitly if they want to keep it.